juriscraper/opinions/united_states/federal_appellate/ca1.py

```diff
@@ -1,124 +1,104 @@
-import re
-from datetime import date, datetime, timedelta
+from datetime import date, datetime
+from typing import Tuple
 from urllib.parse import urlencode
 
-from dateutil.rrule import DAILY, rrule
+from juriscraper.AbstractSite import logger
+from juriscraper.lib.date_utils import make_date_range_tuples
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
-from juriscraper.OpinionSite import OpinionSite
 
+class Site(OpinionSiteLinear):
+    # This URL will show most recent opinions
+    base_url = "https://www.ca1.uscourts.gov/opn/aci"
+    days_interval = 5
+    first_opinion_date = datetime(2003, 3, 23)
 
-class Site(OpinionSite):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.base_url = "http://media.ca1.uscourts.gov/cgi-bin/opinions.pl"
         self.court_id = self.__module__
-        today = date.today()
-        params = urlencode(
-            {
-                "FROMDATE": (today - timedelta(7)).strftime("%m/%d/%Y"),
-                "TODATE": today.strftime("%m/%d/%Y"),
-                "puid": "",
-            }
-        )
-        self.url = f"{self.base_url}/?{params}"
-        # self.url = "http://media.ca1.uscourts.gov/cgi-bin/opinions.pl/?TODATE=06%2F24%2F1993&puid=&FROMDATE=05%2F25%2F1993"
-        self.interval = 30
-        self.back_scrape_iterable = [
-            i.date()
-            for i in rrule(
-                DAILY,
-                interval=self.interval,
-                dtstart=date(1992, 1, 1),
-                until=date(2016, 1, 1),
-            )
-        ]
-
-    def _get_case_names(self):
-        return [
-            e.strip()
-            for e in self.html.xpath(
-                "//tr[position() > 1]/td[4]/text()[contains(., 'v.')]"
-            )
-        ]
-
-    def _get_download_urls(self):
-        return [
-            e for e in self.html.xpath("//tr[position() > 1]/td[2]//@href")
-        ]
+        self.url = self.base_url
+        self.make_backscrape_iterable(kwargs)
 
-    def _get_case_dates(self):
-        dates = []
-        for s in self.html.xpath("//tr[position() > 1]/td[1]//text()"):
-            s = s.replace(r"\t", "").replace(r"\n", "").strip()
-            if s == "1996/05/32":
-                s = "1996/05/30"  # My life is thus lain to waste.
-            dates.append(datetime.strptime(s.strip(), "%Y/%m/%d").date())
-        return dates
+    def _process_html(self):
+        for row in self.html.xpath("//tr[position() > 1]"):
+            title = row.xpath("td[2]/a/text()")[0]
+            url = row.xpath("td[2]/a/@href")[0]
+            status = self.get_status_from_opinion_title(title)
+            if status == "Errata":
+                logger.info(
+                    "Skipping Errata file. title: %s URL: %s", title, url
+                )
+                continue
```

Comment:
did we skip errata before? why the change?

Reply:
We didn't. Will correct this, just saw that we actually have a Errata status

Reply:
thanks.